src/features/device-config/device-config.ts

```diff
@@ -49,7 +53,20 @@ export const generateConfig = async (
 			// Older ones have to use the old "user api keys"
 			return await createUserApiKey(req, (await userPromise).id);
 		}
-		return await createProvisioningApiKey(req, app.id);
+
+		const apiKeyOptions: ApiKeyOptions = {};
+
+		// Checking both req.body and req.query given both GET and POST support
+		// Ref: https://github.com/balena-io/balena-api/blob/master/src/routes/applications.ts#L95
+		if (req.body.name) {
+			apiKeyOptions.name = req.body.name as string;
+		}
+
+		if (req.query.name) {
+			apiKeyOptions.name = req.query.name as string;
+		}
```

Comment:
```suggestion
		const keyName: string | undefined = req.body.provisioningKeyName ?? req.query.provisioningKeyName ?? undefined;
		if (typeof keyName === 'string') {
			apiKeyOptions.name = keyName;
		}
```
I strongly feel that the parameter name should be `provisioningKeyName` instead of just `name`, which better explains what that the value will be used for, and make more sense along all the rest supported parameters.
```
export interface ImgConfigOptions {
	network?: 'ethernet' | 'wifi';
	appUpdatePollInterval?: number;
	wifiKey?: string;
	wifiSsid?: string;
	ip?: string;
	gateway?: string;
	netmask?: string;
	deviceType?: string;
	version: string;
}
```

Reply:
@Page- Is the `provisioningKeyName` fione with you on the req.body as well? I can change it back to `provisioningKeyName` along with test cases.

Reply:
@nitishagar to be clear, I only mean using `provisioningKeyName` instead of `name` for the device config endpoint only (where this comment is). For the api key generation endpoints `name` is more appropriate, as Page suggested.

Reply:
Yes I prefer it to be `provisioningKeyName` because otherwise it's ambiguous, it's just in the case where it's eg `POST /provisioning-api-key` where `name` is unambiguous it's better as it also matches eg `POST /named-user-api-key` for a unified interface when creating api keys